simplemma/lemmatizer.py

```diff
@@ -319,36 +320,19 @@ def text_lemmatizer(
 ) -> List[str]:
     """Convenience function to lemmatize a text using a simple tokenizer.
     Returns a list of tokens and lemmata."""
-    lemmata = []
-    last = "."  # beginning is initial
-    for match in simple_tokenizer(text, iterate=True):
-        # lemmatize, simple heuristic for sentence boundary
-        lemmata.append(
-            lemmatize(
-                match[0],
-                lang=lang,
-                greedy=greedy,
-                silent=silent,
-                initial=last in PUNCTUATION,
-            )
-        )
-        last = match[0]
-    return lemmata
+    return list(lemma_iterator(text, lang, greedy, silent))
 
 
 def lemma_iterator(
     text: str,
     lang: Optional[Union[str, Tuple[str]]] = None,
     greedy: bool = False,
     silent: bool = True,
+    tokenizer: Tokenizer = Tokenizer(),
 ) -> Iterator[str]:
     """Convenience function to lemmatize a text using a simple tokenizer.
     Returns a list of tokens and lemmata."""
-    last = "."  # beginning is initial
-    for match in simple_tokenizer(text, iterate=True):
```

Comment:
Here it also used the iterator. So, the list was never used anywhere 😅

Reply:
Same here, it can stay, I just want the simple_tokenizer function to return a list with `findall`, the other code blocks follow another logic, I'm with you on this.